lab_6_pipeline/pipeline.py

```diff
@@ -27,16 +39,40 @@ def __init__(self, path_to_raw_txt_data: pathlib.Path) -> None:
         Args:
             path_to_raw_txt_data (pathlib.Path): Path to raw txt data
         """
+        self.path_to_raw_txt_data = path_to_raw_txt_data
+        self._storage = {}
+        self._validate_dataset()
+        self._scan_dataset()
 
     def _validate_dataset(self) -> None:
         """
         Validate folder with assets.
         """
+        if not self.path_to_raw_txt_data.exists():
+            raise FileNotFoundError
+        if not self.path_to_raw_txt_data.is_dir():
+            raise NotADirectoryError
+        if not any(self.path_to_raw_txt_data.iterdir()):
+            raise EmptyDirectoryError
+        raw_f = list(self.path_to_raw_txt_data.glob("*_raw.txt"))
+        meta_f = list(self.path_to_raw_txt_data.glob("*_meta.json"))
+        sorted_raw_files = sorted(raw_f, key=lambda file: get_article_id_from_filepath(file))
+        sorted_meta_files = sorted(meta_f, key=lambda file: get_article_id_from_filepath(file))
+        for i, (raw, meta) in enumerate(zip(sorted_raw_files, sorted_meta_files)):
+            if i+1 != get_article_id_from_filepath(raw) or i+1 != get_article_id_from_filepath(meta):
```

Comment:
Consider passing a special argument to `enumerate` to avoid adding 1 at every step

Reply:
fixed